packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs

```diff
@@ -0,0 +1,374 @@
+            None,
+            None,
+            configurables,
+        )
+        .await;
+
+        let proxy = BridgeProxy::new(bridge.contract_id().clone(), wallet.clone());
+
+        let error_receipt = proxy
+            .methods()
+            ._proxy_change_owner(new_owner.into())
+            .with_contract_ids(&[proxy_id.into()])
+            .call()
+            .await
+            .unwrap_err();
+
+        assert!(
+            matches!(error_receipt,
+                Error::Transaction(Reason::Reverted {reason, ..})
+                if reason == "IdentityZero".to_string()
+            ),
+            "Transaction did not revert or reverted with a wrong reason"
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn proxy_change_owner_only_owner() -> anyhow::Result<()> {
+        let mut wallet = create_wallet();
+        let mut mallory = WalletUnlocked::new_random(None);
+
+        let configurables: Option<BridgeFungibleTokenContractConfigurables> = None;
+
+        let (proxy_id, _implementation_contract_id) =
+            get_contract_ids(&wallet, configurables.clone());
+
+        let wallet_funds = (DEFAULT_COIN_AMOUNT, AssetId::default());
+
+        let (_, bridge, _) = setup_environment(
+            &mut wallet,
+            vec![wallet_funds],
+            vec![],
+            None,
+            None,
+            configurables,
+        )
+        .await;
+
+        let _ = wallet
+            .transfer(
+                mallory.address(),
+                DEFAULT_COIN_AMOUNT / 2,
+                Default::default(),
+                Default::default(),
+            )
+            .await?;
+
+        let provider = wallet.provider().clone().unwrap().clone();
+        mallory.set_provider(provider);
+
+        let proxy = BridgeProxy::new(bridge.contract_id().clone(), mallory.clone());
+
+        let error_receipt = proxy
+            .methods()
+            ._proxy_change_owner(mallory.address().into())
+            .with_contract_ids(&[proxy_id.into()])
+            .call()
+            .await
+            .unwrap_err();
+
+        assert!(
+            matches!(error_receipt,
+                Error::Transaction(Reason::Reverted {reason, ..})
+                if reason == "NotOwner".to_string()
+            ),
+            "Transaction did not revert or reverted with a wrong reason"
+        );
+
+        let owner = proxy
+            .methods()
+            ._proxy_owner()
+            .with_contract_ids(&[proxy_id.into()])
+            .simulate()
+            .await?
+            .value;
+
+        assert!(
+            matches!(owner,
+                State::Initialized(fuels::types::Identity::Address(address))
+                if address == wallet.address().clone().into()
+            ),
+            "Ownership was not initialized or owner is not the expected address"
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn proxy_set_target() -> anyhow::Result<()> {
```

Comment:
Can you add an only_owner version of this test/ check in this test that a non-owner calling the method errors

Reply:
Done